syntax = "proto3";

import "Uniform.proto";

package frame.proto;

// Description of an effect that can be used as a 2D effect on a rendering or
// as a shader for material.
// Next 9
message Program {
	// Name of the effect.
	string name = 1;
	// Input texture names.
	repeated string input_texture_names = 3;
	// Output texture names.
	repeated string output_texture_names = 4;
	// Input Scenes name.
	string input_scene_name = 5;
	// Shader name (used to find the shader).
	string shader = 6;
	// Additionnal parameters for the shader.
	repeated Uniform parameters = 7;
}

// Description of an effect file.
// Next 2
message ProgramFile {
	// Contain many programs.
	repeated Program programs = 1;
}
